### ISSUE TITLE
Percentiles calculation explained?

### ISSUE THREAD

**Issue by asaff1** [bojand/ghz#215]
**Command line arguments / config**
./ghz --insecure --total 15 --connections 1 --concurrency 10 --timeout 0 --proto ./data/svc.proto --call svc.v1.Service.Start. --binary-file data.pb --duration 0 --skipFirst 0 -O json localhost:50051

**Describe the bug**
I believe the percentiles are not accurate. See example for median.
This is the `details` section of the output json:  
```
"details": [
      {
         "timestamp":"2020-08-19T15:45:58.495351217+03:00",
         "latency":1042165691,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:58.501101627+03:00",
         "latency":1047885273,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:58.505083345+03:00",
         "latency":1051897867,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:58.515618616+03:00",
         "latency":1068984262,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:58.529694695+03:00",
         "latency":1083060826,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:59.506724066+03:00",
         "latency":1011303003,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:59.512665534+03:00",
         "latency":996989754,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:59.522287957+03:00",
         "latency":992551499,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:59.530468266+03:00",
         "latency":1029325988,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:45:59.544120387+03:00",
         "latency":1038985451,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:46:00.537823106+03:00",
         "latency":993612515,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:46:00.538416936+03:00",
         "latency":1007881907,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:46:00.543929337+03:00",
         "latency":1031228969,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:46:00.56012676+03:00",
         "latency":1053347631,
         "error":"",
         "status":"OK"
      },
      {
         "timestamp":"2020-08-19T15:46:00.563120347+03:00",
         "latency":1040774187,
         "error":"",
         "status":"OK"
      }
   ]
```
These are the percentiles:
```
"latencyDistribution":[
      {
         "percentage":10,
         "latency":996989754
      },
      {
         "percentage":25,
         "latency":1011303003
      },
      {
         "percentage":50,
         "latency":1040774187
      },
      {
         "percentage":75,
         "latency":1053347631
      },
      {
         "percentage":90,
         "latency":1083060826
      },
      {
         "percentage":0,
         "latency":0
      },
      {
         "percentage":0,
         "latency":0
      }
   ]
```

**Expected behavior**
The sorted latency values are:
`[ 992551499,  993612515,  996989754, 1007881907, 1011303003,
       1029325988, 1031228969, 1038985451, 1040774187, 1042165691,
       1047885273, 1051897867, 1053347631, 1068984262, 1083060826]`

**There are 15 items, the median is is the 8th item which is 1038985451 (and not 1040774187 as in the output.).**

Using numpy:
`np.percentile(l, [10, 25, 50, 75, 90], interpolation='higher') `
Output:
`>> array([ 996989754, 1011303003, 1038985451, 1051897867, 1068984262])`
Which is slightly different for the 50% and the 75%.

Second, Why I the two last items in the array with percentage=0?

**Environment**
 - OS: Ubuntu 18.04
 - ghz: 0.56.0

**Comment by bojand** [comment]
Hello, the percentile calculation wasn't completely correct. This has been addressed in [0.57.0](https://github.com/bojand/ghz/releases/tag/v0.57.0) to use [nearest rank](https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method) method to calculate percentiles.

**Comment by asaff1** [comment]
Thanks. I did a quick check and it looks better. I think there's still a slight inaccuracy, I will post an update if needed.
The two zeros in the `latencyDistribution` which broke ghz-web compare option are gone.

I think the histogram also need a slight fix. On every run, the first bucket always counts as 1, which does not provide any information (it is simply the minimum value).
I think the output should be exactly 10 bins (currently it is 11 bins), so I would drop the first bucket, and each `"mark"` should be only the right edge of each bucket in the histogram.
